kalpsdk/transaction_context.go

```diff
@@ -10,7 +12,29 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+type HistoryQueryResult struct {
+	TxId      string
+	Timestamp time.Time
+	Record    string
+	IsDelete  bool
+}
+
 type TransactionContextInterface interface {
+	// Initialize initializes the smart contract owner by setting the owner ID.
+	// It checks if there's already an owner, and if not, it sets the caller's userID as the owner ID.
+	Initialize() error
+
+	// IsSmartContractOwner checks if the caller is the owner of the smart contract
+	// by comparing the userID of the caller with the stored smart contract owner ID.
+	IsSmartContractOwner() (bool, error)
+
+	// TransferOwner transfers ownership of the smart contract to the caller.
```

Comment:
@chiragp2e Please update comment to indicate ownership change to new owner.

Reply:
done